### DISCUSSION TITLE
Set two notifications with sounds but only 1 displays, and DisplayAlert() not able to use, please help

### DISCUSSION THREAD

**Discussion by Cinmay2014** [thudugala/Plugin.LocalNotification#392] [Q&A]
I am learning Xamarin forms and trying to use Plugin.localnotification.  I created two notification channels with different sounds: channel1 for the sound:goodnight and channel2 for the sound:goodmorning. But Notification1 displayed goodmoring sound and notification2 displayed a default sound. By the way, I am not able to use DisplayAlert(). Please help.
Thanks in advance.

My MainPageViewModel.cs shows below:
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using System.Windows.Input;
using Xamarin.Forms;
using Plugin.LocalNotification;
using Xamarin.Essentials;
using Plugin.LocalNotification.AndroidOption;
using Plugin.LocalNotification.Json;
using System.Threading.Tasks;
using Plugin.LocalNotification.EventArgs;
using System.IO;
using Xamarin.Forms.Internals;

namespace localnotice
{
    internal class MainPageViewModel : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        // notification1
        private int _notificationId1 = 88;
       
        private string _title1;
        public string Title1
        {
            get => _title1;
            set
            {
                _title1 = value;
                OnPropertyChanged(nameof(Title1));
            }
        }

        private string _message1;
        public string Message1
        {
            get => _message1;
            set
            {
                _message1 = value;
                OnPropertyChanged(nameof(Message1));
            }
        }

        private TimeSpan _notifyTime1;
        public TimeSpan NotifyTime1
        {
            get => _notifyTime1;
            set
            {
                _notifyTime1 = value;
                OnPropertyChanged(nameof(NotifyTime1));
            }
        }

        // notification2
        private int _notificationId2 = 89;
        private string _title2;
        public string Title2
        {
            get => _title2;
            set
            {
                _title2 = value;
                OnPropertyChanged(nameof(Title2));
            }
        }

        private string _message2;
        public string Message2
        {
            get => _message2;
            set
            {
                _message2 = value;
                OnPropertyChanged(nameof(Message2));
            }
        }

        private TimeSpan _notifyTime2;
        public TimeSpan NotifyTime2
        {
            get => _notifyTime2;
            set
            {
                _notifyTime2 = value;
                OnPropertyChanged(nameof(NotifyTime2));
            }
        }

        private bool _isRepeatEnabled;
        public bool IsRepeatEnabled
        {
            get => _isRepeatEnabled;
            set
            {
                _isRepeatEnabled = value;
                OnPropertyChanged(nameof(IsRepeatEnabled));
            }
        }

        private string _repeatInterval;
        public string RepeatInterval
        {
            get => _repeatInterval;
            set
            {
                _repeatInterval = value;
                OnPropertyChanged(nameof(RepeatInterval));
            }
        }

        public List<string> RepeatOptions { get; }
        public ICommand SetNotificationCommand => new Command(SetNotification);

        public MainPageViewModel()
        {
            // Initialize properties for notification1
            Title1 = string.Empty;
            Message1 = string.Empty;
            NotifyTime1 = DateTime.Now.TimeOfDay;
            // Initialize properties for notification2
            Title2 = string.Empty;
            Message2 = string.Empty;
            NotifyTime2 = DateTime.Now.TimeOfDay;

            IsRepeatEnabled = false;
            RepeatInterval = string.Empty;
            RepeatOptions = new List<string> { "Daily", "Weekly", "Monthly" };
            LocalNotificationCenter.Current.NotificationReceived += ShowCustomAlertFromNotification;
            LocalNotificationCenter.Current.NotificationActionTapped += Current_NotificationActionTapped;

        }

        private void Current_NotificationActionTapped(NotificationActionEventArgs e)
        {
            if (e.IsDismissed)
            {
                Device.BeginInvokeOnMainThread(() =>
                {
                  //DisplayAlert(e.Request.Title, "User Dismissed Notification", "OK");
                });
                return;
            }
        }

        private void ShowCustomAlertFromNotification(NotificationEventArgs e)
        {
          //
        }

        protected void OnPropertyChanged(string propertyName)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        private void SetNotification()
        {
            ScheduleNotification(Title1, Message1, NotifyTime1, _notificationId1, "goodnight", "my_channel_01");
            ScheduleNotification(Title2, Message2, NotifyTime2, _notificationId2, "goodmorning", "my_channel_02");
            
        }

        //private void ScheduleNotification(string title, string message, TimeSpan notifyTime, int notificationId, string androidSound, string androidSoundExt, string channelId)
        private void ScheduleNotification(string title, string message, TimeSpan notifyTime, int notificationId, string androidSound, string channelId)
        {
            System.Diagnostics.Debug.WriteLine(channelId);
            DateTime notifyDateTime = DateTime.Today.Add(notifyTime);
            TimeSpan repeatInterval = TimeSpan.Zero;

            if (IsRepeatEnabled)
            {
                switch (RepeatInterval)
                {
                    case "Daily":
                        repeatInterval = TimeSpan.FromDays(1);
                        break;
                    case "Weekly":
                        repeatInterval = TimeSpan.FromDays(7);
                        break;
                    case "Monthly":
                        repeatInterval = TimeSpan.FromDays(30);
                        break;
                    default:
                        throw new ArgumentException("Invalid repeat interval");
                }
            }

            // Schedule the notification
            var notification = new NotificationRequest
            {
                NotificationId = notificationId,
                Title = title,
                Description = message,
                Schedule = new NotificationRequestSchedule
                {
                    NotifyTime = notifyDateTime,
                    NotifyRepeatInterval = repeatInterval
                },
                Android = { ChannelId = channelId }
                
            };

            LocalNotificationCenter.Current.Show(notification);
        }

    }
}

$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
MainActivity.cs as below:
using System;

using Android.App;
using Android.Content.PM;
using Android.Runtime;
using Android.OS;
using Plugin.LocalNotification;
using Plugin.LocalNotification.AndroidOption;
using Android.Content;
using static Java.Util.Jar.Attributes;

namespace localnotice.Droid
{
    [Activity(Label = "localnotice", Icon = "@mipmap/icon", Theme = "@style/MainTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation | ConfigChanges.UiMode | ConfigChanges.ScreenLayout | ConfigChanges.SmallestScreenSize )]
    public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
            global::Xamarin.Forms.Forms.Init(this, savedInstanceState);
            // Initialize NotificationCenter
            LocalNotificationCenter.CreateNotificationChannel(
                //channel 01 for night
                new NotificationChannelRequest 
            {
                Id = $"my_channel_01",
                Name = "General",
                Description = "General",
                Sound = Resource.Raw.goodnight.ToString()
                });

            LocalNotificationCenter.CreateNotificationChannel(
                //channel 02 for morning
                    new NotificationChannelRequest
                    {
                        Id = $"my_channel_02",
                        Name = "Special",
                        Description = "Special",
                        Sound = Resource.Raw.goodmorning.ToString()
                    });
            LoadApplication(new App());
            LocalNotificationCenter.NotifyNotificationTapped(Intent);
        }
        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
        {
            Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }

        protected override void OnNewIntent(Intent intent)
        {
            LocalNotificationCenter.NotifyNotificationTapped(intent);
            base.OnNewIntent(intent);
        }
    }
}

**Comment by thudugala** [comment]
@Cinmay2014 Can you please attach a sample project?
